app/controllers/dashboard/advisers_controller.rb

```diff
@@ -7,10 +7,20 @@ def index
       @trading_names = @firm.trading_names
     end
 
+    def new
+      @firm = Firm.find(params[:firm_id])
+      @adviser = advisers.build
+    end
+
     private
 
     def principal
       current_user.principal
     end
+
+    def advisers
+      Firm.find_by(id: params[:firm_id], fca_number: principal.fca_number)
+        .advisers
```

Comment:
That won't work — because `principal.firm` returns a Firm object, not a relation. So I guess you meant: `principal.firm.advisers`?

We don't do that because while the Principal model claims seems to only have one firm, that's a bit misleading. The Principal only has one Firm directly, but that Firm can have its own child Firms. [See here.](https://github.com/moneyadviceservice/mas-rad_core/blob/master/app/models/firm.rb#L34) Each of these will have the same `fca_number`. It took us a while to get our heads around.

So we need to specify the ID to make sure we get the one we're interested in. We need to specify the `fca_number` so the principal can only access their own firms.